tutorial/tests/test_object_oriented_programming.py

```diff
@@ -2,6 +2,221 @@
 
 import pytest
 
+#
+# Example 1: Person
+#
+
+
+def reference_oop_person(first_name: str, last_name: str):
+    class Person:
+        """A class representing a person with first name and last name"""
+
+        def __init__(self, first_name: str, last_name: str):
+            self.first_name = first_name
+            self.last_name = last_name
+
+    return Person(first_name, last_name)
+
+
+def validate_oop_person(solution_result):
+    params = list(vars(solution_result))
+    assert (
+        type(solution_result).__name__ == "Person"
+    ), "The class should be named 'Person'."
+    assert len(params) == 2, "The class should have 2 attributes."
+    assert (
+        "first_name" in params and "last_name" in params
+    ), "The class attributes should be 'first_name' and 'last_name'."

```

Comment:
This is a bit fragile: `vars` requires an object with the `__dict__` method, and if the user returns a string, then we get a `TypeError` which will confuse them.

My suggestion (although it's more verbose):

```suggestion
def validate_oop_person(solution_result):
   # Verify it's not a primitive type
    assert not isinstance(solution_result, (str, int, float, bool, list, dict, tuple)), "Solution must return a class instance, not a primitive"
    # Verify that's a custom class, not a built-in
    assert type(solution_result).__module__ != "builtins", "Solution must return a class instance, not a built-in type"
    # Now the name
    assert type(solution_result).__name__ == "Person", "The class name should be 'Person'"
    
    # Then check the attributes
    try:
        params = list(vars(solution_result))
    except TypeError:
        raise AssertionError("Solution must be a proper class instance with attributes")
    
    assert len(params) == 2, "The class should have 2 attributes"
    assert "first_name" in params and "last_name" in params, \
        "The class attributes should be 'first_name' and 'last_name'"
```

Ideally, using `inspect` is even better, but I think this is good enough.

Reply:
Just for _fun_, here's an alternative with `inspect`. The only benefit is that it would work with inheritance (but it's not the case here).

```python
import inspect

# whatever comes before ...

def validate_oop_person(solution_result):
    # First verify we have an object instance, not a primitive
    assert not isinstance(solution_result, (str, int, float, bool, list, dict, tuple)), "Solution must return a class instance, not a primitive type"
    
    # Verify it's a custom class (not a built-in type)
    assert not type(solution_result).__module__ == 'builtins', "Solution must return a custom class instance, not a built-in type"
    
    # Verify the class name
    assert type(solution_result).__name__ == "Person", "The class should be named 'Person'"
    
    # Get attributes using inspect
    attributes = inspect.getmembers(solution_result, lambda a: not inspect.isroutine(a))
    
    # Filter out unwanted attributes
    attributes = [a[0] for a in attributes if not a[0].startswith('_')]
    
    assert len(attributes) == 2, "The class should have 2 attributes"
    assert "first_name" in attributes and "last_name" in attributes, "The class attributes should be 'first_name' and 'last_name'"
```

Reply:
Thanks for the suggestion @edoardob90, it has been addressed with this commit: 56ba50bb4ac3e18bdf2ffd64b4fb18e48ad55929